### PR TITLE
Buncha instances

### PR DESCRIPTION
**Pull request by danbornside**

### CODE REVIEW

File: src/Reflex/Patch.hs [hunk 1]
```diff
@@ -44,3 +52,39 @@ instance (Ord k, Group q) => Group (MonoidalMap k q) where
...
+instance Group (f (g a)) => Group ((f :.: g) a) where
+  negateG (Comp1 xs) = Comp1 (negateG xs)
+  Comp1 xs ~~ Comp1 ys = Comp1 (xs ~~ ys)
+
+instance Additive (f (g a)) => Additive ((f :.: g) a) where
+
+instance (Group (f a), Group (g a)) => Group ((f :*: g) a) where
+  negateG (a :*: b) = negateG a :*: negateG b
+  (a :*: b) ~~ (c :*: d) = (a ~~ c :*: b ~~ d)
+instance (Additive (f a), Additive (g a)) => Additive ((f :*: g) a) where
```
Comment (3noch) [inline src/Reflex/Patch.hs]: You can elide the `where` to make it clearer that you didn't intend to put anything here.

**Comment by 3noch** [comment]
We *could* just use the hackage package for this and get most of these: http://hackage.haskell.org/package/groups-0.4.1.0/docs/Data-Group.html

File: src/Reflex/Query/Class.hs [hunk 2]
```diff
@@ -50,6 +55,25 @@ instance (Ord k, Query v) => Query (MonoidalMap k v) where
...
+  type QueryResult () = ()
+  crop _ _ = ()
+
+instance Query Void where
+  type QueryResult Void = ()
+  crop = absurd
+
+#if MIN_VERSION_base(4,12,0)
+-- | We can lift queries into monoidal containers.
+-- But beware of Applicatives who's monoid is different from (pure mempty, liftA2 mappend)
```
Comment (alexfmpe) [inline src/Reflex/Query/Class.hs]: whose?

**Comment by Ericson2314** [comment]
Just need to fix GHC <= 8.2.

**Comment by Ericson2314** [comment]
Ugh....hlint....

**Pull request merged**